lib/reader/src/parser.rs

```diff
@@ -959,9 +959,32 @@ impl<'a> Parser<'a> {
         ctx.map.def_value(vx, value, &vx_location)
     }
 
+    fn parse_value_location(&mut self, ctx: &Context, isa: &TargetIsa) -> Result<ValueLoc> {
+        let reginfo = isa.register_info();
+        match self.token() {
+            Some(Token::StackSlot(src_num)) => {
+                self.consume();
+                if let Some(ss) = ctx.map.get_ss(src_num) {
+                    Ok(ValueLoc::Stack(ss))
+                } else {
+                    err!(self.loc,
+                         "attempted to use undefined stack slot ss{}",
+                         src_num)
+                }
+            }
+            Some(Token::Name(name)) => {
+                self.consume();
+                reginfo.parse_regunit(name)
+                    .map(ValueLoc::Reg)
+                    .ok_or(self.error("invalid register value location"))
+            }
+            _ => err!(self.loc, "invalid value location"),
+        }
+    }
+
     fn parse_instruction_encoding(&mut self,
                                   ctx: &Context)
-                                  -> Result<(Option<Encoding>, Option<Vec<&'a str>>)> {
+                                  -> Result<(Option<Encoding>, Option<Vec<ValueLoc>>)> {
```

Comment:
I think there is a difference, eg you can have a `return` instruction which specifies no locations because it has no results, and you can have an `iadd` instruction that just omits the encoding information for its 1 result. Both of those would have `vec.len() == 0` but should be treated differently.